sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -28,30 +28,44 @@ impl BearerTokenCredentialPolicy {
         Self {
             credential,
             scopes: scopes.into_iter().map(|s| s.into()).collect(),
-            access_token: Arc::new(Mutex::new("UNSET".to_string())), // Need a better default here or use Option
-            last_refresh_time: Arc::new(Mutex::new(0)), // Need a better default here aswell
+            access_token: Arc::new(Mutex::new(None)),
+            last_refresh_time: Arc::new(Mutex::new(None)),
         }
     }
 
-    fn update_data(&self, new_access_token: &str) {
-        let mut my_access_token = self.access_token.lock().unwrap();
-        *my_access_token = new_access_token.to_string();
-        let mut my_last_refresh_time = self.last_refresh_time.lock().unwrap();
-        *my_last_refresh_time = OffsetDateTime::now_utc().unix_timestamp();
+    fn scopes(&self) -> Vec<&str> {
+        self.scopes
+            .iter()
+            .map(String::as_str)
+            .collect::<Vec<&str>>()
+    }
+
+    fn update_token(&self, new_access_token: &AccessToken) {
+        let mut access_token = self.access_token.lock().unwrap();
+        *access_token = Some(new_access_token.clone());
+        let mut last_refresh_time = self.last_refresh_time.lock().unwrap();
+        *last_refresh_time = Some(OffsetDateTime::now_utc().unix_timestamp());
     }
 
-    fn should_refresh(&self) -> bool {
-        let current_time = OffsetDateTime::now_utc().unix_timestamp();
-        let my_last_refresh_time = self.last_refresh_time.lock().unwrap();
-        if current_time - *my_last_refresh_time > DEFAULT_REFRESH_TIME {
-            return true;
+    fn is_token_expired(&self) -> bool {
+        let last_refresh_time = self.last_refresh_time.lock().unwrap();
+        match *last_refresh_time {
+            Some(refresh_time) => {
+                OffsetDateTime::now_utc().unix_timestamp() - refresh_time > DEFAULT_REFRESH_TIME
+            }
+            None => true,
         }
-        return false;
     }
 
-    fn access_token(&self) -> String {
-        let my_access_token = self.access_token.lock().unwrap();
-        (*my_access_token.clone()).to_string()
+    fn access_token(&self) -> Result<String, &'static str> {
```

Comment:
Only return `azure_core::Result<T>` from our APIs.

Reply:
...unless, in some cases, `io::Result`, `fmt::Result`, or other is more appropriate. In this case it's not.